package/MDAnalysis/coordinates/base.py

```diff
@@ -1014,16 +1040,16 @@ def add_auxiliary(self, auxname, auxdata, format=None, **kwargs):
         Auxiliary data is assumed to be time-ordered, with no duplicates. See
         the :ref:`Auxiliary API`.
         """
-        if auxname in self.aux_list:
-            raise ValueError("Auxiliary data with name {name} already "
-                             "exists".format(name=auxname))
-        if isinstance(auxdata, AuxReader):
-            aux = auxdata
-            aux.auxname = auxname
+        if auxdata is None:
+            raise AttributeError("No input `auxdata` specified, but it needs "
```

Comment:
AttributeError does not make sense when there's input missing. Should be `ValueError`.

See https://docs.python.org/3/library/exceptions.html#ValueError and discussion at https://docs.python.org/3/library/exceptions.html#TypeError